refinery/core/tests.py

```diff
@@ -2295,3 +2296,49 @@ def test_create_workflow_engine_bad_group_name(self):
                 str(galaxy_instance.id),
                 "non-existent group name"
             )
+
+
+class TestIsAbsoluteURL(TestCase):
+
+    def test_is_absolute_url(self):
+        self.assertTrue(is_absolute_url('http://example.org'))
+        self.assertTrue(is_absolute_url('ftp://example.org'))
+
+    def test_is_not_absolute_url(self):
+        self.assertFalse(is_absolute_url('/path/to/file'))
+        self.assertFalse(is_absolute_url('example.org'))
+
+    def test_is_absolute_url_with_empty_input(self):
+        self.assertFalse(is_absolute_url(None))
+        self.assertFalse(is_absolute_url(''))
+
+    def test_get_absolute_url_with_absolute_url(self):
+        self.assertEqual(get_absolute_url('http://example.org'),
+                         'http://example.org')
+
+
+@override_settings(REFINERY_URL_SCHEME='http')
+class TestGetAbsoluteURL(TestCase):
+
+    def setUp(self):
+        self.current_site = Site.objects.get_or_create(name='Test',
+                                                       domain='example.org')[0]
+
+    def test_get_absolute_url_with_relative_url(self):
+        with override_settings(SITE_ID=self.current_site.id):
+            self.assertEqual(get_absolute_url('/path/to/file'),
+                             'http://example.org/path/to/file')
+
+    def test_get_absolute_url_with_absolute_url(self):
+        with override_settings(SITE_ID=self.current_site.id):
+            self.assertEqual(get_absolute_url('http://example.org'),
+                             'http://example.org')
+
+    def test_get_absolute_url_with_empty_input(self):
+        with override_settings(SITE_ID=self.current_site.id):
+            self.assertEqual(get_absolute_url(None), None)
+            self.assertEqual(get_absolute_url(''), '')
```

Comment:
Is this the behavior you want? I worry that folks could be surprised when the result of `get_absolute_url` does not pass `is_absolute_url`. Maybe it should throw an exception?

Reply:
Good question. I would have done this but since I was refactoring, I've kept the function signature as is. Raising an exception requires exception handling in every use case. I'll open an issue for this.

Reply:
See #2636